### PR TITLE
updated to Event Grid trigger

### PR DESCRIPTION
**Pull request by tfitzmac**

### CODE REVIEW

File: samples/e2e/EventHubsCaptureEventGridDemo/FunctionEGDWDumper/Function1.cs [hunk 1]
```diff
@@ -10,70 +13,38 @@
...
-        /// c. Execute the function in Azure portal as a one-off by selecting POST and providing the json body you copied
-        /// d. This should create some rows in the data warehouse.
-        ///    This ensures that the Azure Function is dumping data correctly to the data warehouse
-        /// e. Next, create a new EventGrid subscription name to Azure Functions
-        ///    This should automatically populate data to the data warehouse every time the Avro blob is created by Event Hubs Capture.
-        /// </summary>
-        /// <returns></returns>
-        [FunctionName("HttpTriggerCSharp")]
-        public static async Task<HttpResponseMessage> Run([HttpTrigger(AuthorizationLevel.Function, "get", "post", Route = null)]HttpRequestMessage req, TraceWriter log)
+        [FunctionName("Function1")]
```
Comment (ShubhaVijayasarathy) [inline samples/e2e/EventHubsCaptureEventGridDemo/FunctionEGDWDumper/Function1.cs]: Can we provide descriptive function name?

File: samples/e2e/EventHubsCaptureEventGridDemo/FunctionEGDWDumper/Function1.cs [hunk 2]
```diff
@@ -10,70 +13,38 @@
...
+        public static void Run([EventGridTrigger]JObject eventGridEvent, TraceWriter log)
```
Comment (ShubhaVijayasarathy) [inline samples/e2e/EventHubsCaptureEventGridDemo/FunctionEGDWDumper/Function1.cs]: Also, can we add comments describing the steps? please who download will get better perspective (refer to the previous code)

Comment (ShubhaVijayasarathy) [reply samples/e2e/EventHubsCaptureEventGridDemo/FunctionEGDWDumper/Function1.cs]: As a best practice, lets add Async methods and exception handling as well

File: samples/e2e/EventHubsCaptureEventGridDemo/WindTurbineDataGenerator/Program.cs [hunk 3]
```diff
@@ -15,7 +15,7 @@ internal class Program
         private const string EventHubConnectionString =
             "[provide the EH connection string]";
 
-        private const string EventHubName = "[provide the EH name]";
+        private const string EventHubName = "hubdatamigration";
```
Comment (ShubhaVijayasarathy) [inline samples/e2e/EventHubsCaptureEventGridDemo/WindTurbineDataGenerator/Program.cs]: you may want to remove any names and add just the place holders

File: samples/e2e/EventHubsCaptureEventGridDemo/FunctionEGDWDumper/Function1.cs [hunk 4]
```diff
@@ -10,70 +13,38 @@
...
-            }
+            var uri = new Uri(ehEvent.data.fileUrl);
+            Dump(uri);
         }
 
-        /// <summary>
-        /// Dumps the data from the Avro blob to the data warehouse (DW). 
-        /// Before running this, ensure that the DW has the required <see cref="TableName"/> table created.
-        /// </summary>        
         private static void Dump(Uri fileUri)
```
Comment (ShubhaVijayasarathy) [inline samples/e2e/EventHubsCaptureEventGridDemo/FunctionEGDWDumper/Function1.cs]: Add ample comments

**Comment by tfitzmac** [comment]
@ShubhaVijayasarathy - I believe I have addressed the feedback items.

**Comment by ShubhaVijayasarathy** [comment]
@serkantkaraca  could you please review the code and merge this please.
Thanks!

File: samples/e2e/EventHubsCaptureEventGridDemo/FunctionEGDWDumper/Function1.cs [hunk 5]
```diff
@@ -10,70 +13,55 @@
...
-        /// d. This should create some rows in the data warehouse.
-        ///    This ensures that the Azure Function is dumping data correctly to the data warehouse
-        /// e. Next, create a new EventGrid subscription name to Azure Functions
-        ///    This should automatically populate data to the data warehouse every time the Avro blob is created by Event Hubs Capture.
-        /// </summary>
-        /// <returns></returns>
-        [FunctionName("HttpTriggerCSharp")]
-        public static async Task<HttpResponseMessage> Run([HttpTrigger(AuthorizationLevel.Function, "get", "post", Route = null)]HttpRequestMessage req, TraceWriter log)
+        [FunctionName("EventGridTriggerMigrateData")]
+        public static void Run([EventGridTrigger]JObject eventGridEvent, TraceWriter log)
```
[the remark this answers is not in the document]
Comment (ShubhaVijayasarathy) [reply samples/e2e/EventHubsCaptureEventGridDemo/FunctionEGDWDumper/Function1.cs]: ah ok

**Pull request merged**